src/modules/Credential/containers/IssuedList/useIssuedList.tsx

```diff
@@ -37,43 +57,67 @@ export const useIssuedList = () => {
   const onRevokeClick = () => setOpenModal({ name: 'revoke', open: true });
 
   const onRevokeCredential = async () => {
-    const { error } = await revokeCredentialAdaptor(selectedCredential);
-    if (error) {
-      console.log(error);
-      return;
+    if (selectedCredential) {
+      const { error } = await revokeCredentialAdaptor(selectedCredential);
+      if (error) {
+        console.log(error);
+        return;
+      }
+      const revokedList = currentList.map(list =>
+        list.id === selectedCredential ? { ...list, status: 'REVOKED' as CredentialStatus } : list,
+      );
+      setCurrentCredentialList({ ...currentCredentialList, items: revokedList });
+      setSelectedCredential('');
+      handleCloseModal();
     }
-    const revokedCredential = currentList.map(list =>
-      list.id === selectedCredential ? { ...list, status: 'REVOKED' as CredentialStatus } : list,
-    );
-    setCurrentList(revokedCredential);
-    setSelectedCredential('');
-    handleCloseModal();
   };
 
   const onDeleteClick = () => setOpenModal({ name: 'delete', open: true });
 
   const onDeleteCredential = async () => {
-    const { error } = await deleteCredentialAdaptor(selectedCredential);
-    if (error) {
-      console.log(error);
-      return;
+    if (selectedCredential) {
+      const { error } = await deleteCredentialAdaptor(selectedCredential);
+      if (error) {
+        console.log(error);
+        return;
+      }
+      const filteredList = currentList.filter(list => list.id !== selectedCredential);
+      if (filteredList.length === 0 && page > 1) {
+        setPage(page - 1);
+        onChangePage(page - 1);
```

Comment:
no need to set page, the page is set again in the 'onChangePage' function

FMI: Why didn't you write useEffect for [page] instead of calling onChangePage?

Reply:
Thanks for your comment, I forgot to remove `setPage` and applied it everywhere.

FYI: when you can access the managing page outside of the `<Pagination />` component, you can make the API request by changing the page easily.
with using of `useEffect` you should put its dependency on `page` (with every change of the `page` state you already have access to changing the `page` state. no need `useEffect`)